### ISSUE TITLE
[FR] include Nebula as 3rd option next to wg and ovpn. 

### ISSUE THREAD

**Issue by krouter** [pivpn/pivpn#1228]
**Is your feature request related to a problem? Please describe.**
I've been using pivpn for over a year now. Very happy on wireguard. Except, my isp switched to CGNAT and ever since I can't access my plex server at home, when away from home. I don't like installing zerotier or other solutions since they are not transparent enough and it doesn't feel FOSS enough. 

**Describe the solution you'd like**
Nebula (by slack) is something I've recently tried and got it working. It's straight forward with Google cloud free tier and minimal bandwidth used only for handshakes. All peer to peer routing is directly between the clients. This is an excellent candidate for pivpn team to wrap into their offering just like you've done with wireguard. 
I wouldn't have dared to use wg if it wasn't for your effort to dumb it down for us. 

**Describe alternatives you've considered**
Zerotier but it's not FOSS enough. 

**Additional context**

**Comment by 4s3ti** [comment]
Hi @krouter  thanks for your request.. however this is completely out of pivpn scope ... 

That would be almost worth of a new PiNebula script ... 

Also Nebula is fundamentally different than OpenVPN or Wireguard, this said PiVPN is meant to be exactly what it is ... a simple way of installing a VPN and i rather have it doing only this task and doing it well than adding more complexity layers on it.

**Comment by TinCanTech** [comment]
@4s3ti Given time and motivation the idea of a PiNebula script sounds quite feasible.

@krouter Question: Have you setup your **Lighthouse** node ?

After reading about Nebula, it looks very much like it is a mixture of a _mesh network_ and essentially OpenVPN. The Nebula _Lighthouse_ being equivalent to an OpenVPN Server.

The Lighthouse being required to reach "client" nodes behind difficult firewalls.

**Comment by 4s3ti** [comment]
> @4s3ti Given time and motivation the idea of a PiNebula script sounds quite feasible.

That resumes pretty much everything ... but motivation i'd say is the least ... Time, and people are my main concern, also for now i don't want to add any extra layers of complexity to pivpn.
